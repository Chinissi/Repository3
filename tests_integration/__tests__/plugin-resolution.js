"use strict";

const runPrettier = require("../runPrettier");
const EOL = require("os").EOL;

describe("automatically loads 'prettier-plugin-*'", () => {
  runPrettier("plugins/automatic", ["file.txt", "--parser=foo"]).test({
    stdout: "foo+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("automatically loads '@prettier/plugin-*'", () => {
  runPrettier("plugins/automatic", ["file.txt", "--parser=bar"]).test({
    stdout: "bar+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("automatically loads 'prettier-plugin-*' from --plugin-search-dir (same as autoload dir)", () => {
  runPrettier("plugins/automatic", [
    "file.txt",
    "--parser=foo",
    `--plugin-search-dir=.`
  ]).test({
    stdout: "foo+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("automatically loads '@prettier/plugin-*' from --plugin-search-dir (same as autoload dir)", () => {
  runPrettier("plugins/automatic", [
    "file.txt",
    "--parser=bar",
    `--plugin-search-dir=.`
  ]).test({
    stdout: "bar+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("automatically loads 'prettier-plugin-*' from --plugin-search-dir (different to autoload dir)", () => {
  runPrettier("plugins", [
    "automatic/file.txt",
    "--parser=foo",
    `--plugin-search-dir=automatic`
  ]).test({
    stdout: "foo+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("automatically loads '@prettier/plugin-*' from --plugin-search-dir (different to autoload dir)", () => {
  runPrettier("plugins", [
    "automatic/file.txt",
    "--parser=bar",
    `--plugin-search-dir=automatic`
  ]).test({
    stdout: "bar+contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("does not crash when --plugin-search-dir does not contain node_modules", () => {
  runPrettier("plugins/extensions", [
    "file.foo",
    "--plugin=./plugin",
    `--plugin-search-dir=.`
  ]).test({
    stdout: "!contents" + EOL,
    stderr: "",
    status: 0,
    write: []
  });
});

describe("crashes when one of --plugin-search-dir does not exist", () => {
  runPrettier("plugins/automatic", [
    "file.txt",
    "--parser=foo",
    `--plugin-search-dir=non-existing-dir`,
    `--plugin-search-dir=.`
  ]).test({
    stdout: "",
    stderr: "non-existing-dir does not exist or is not a directory",
    status: 1,
    write: []
  });
});
