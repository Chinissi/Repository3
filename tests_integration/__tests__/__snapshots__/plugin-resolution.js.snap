// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automatically loads '@prettier/plugin-*' (formatCalls) 1`] = `1`;

exports[`automatically loads '@prettier/plugin-*' from --plugin-search-dir (different to autoload dir) (formatCalls) 1`] = `1`;

exports[`automatically loads '@prettier/plugin-*' from --plugin-search-dir (same as autoload dir) (formatCalls) 1`] = `1`;

exports[`automatically loads 'prettier-plugin-*' (formatCalls) 1`] = `1`;

exports[`automatically loads 'prettier-plugin-*' from --plugin-search-dir (different to autoload dir) (formatCalls) 1`] = `1`;

exports[`automatically loads 'prettier-plugin-*' from --plugin-search-dir (same as autoload dir) (formatCalls) 1`] = `1`;

exports[`crashes when one of --plugin-search-dir does not exist (formatCalls) 1`] = `0`;

exports[`does not crash when --plugin-search-dir does not contain node_modules (formatCalls) 1`] = `1`;

exports[`loads --plugin by bespoke plugin name (assuming it is installed in cwd) (formatCalls) 1`] = `1`;

exports[`loads --plugin by its relative path (formatCalls) 1`] = `1`;

exports[`loads --plugin by its relative path without leading ./ (formatCalls) 1`] = `1`;

exports[`loads --plugin by relative path to its directory (assuming index.js) (formatCalls) 1`] = `1`;

exports[`loads --plugin by relative path to its directory without leading ./ (assuming index.js) (formatCalls) 1`] = `1`;
